### PR TITLE
Add test-only APIController constructor for Vertex AI unit tests

### PR DESCRIPTION
**Pull request by tanzimfh**

### CODE REVIEW

File: common/build.gradle.kts [hunk 1]
```diff
@@ -95,6 +95,7 @@ dependencies {
     testImplementation("io.kotest:kotest-assertions-jvm:4.0.7")
     testImplementation("io.kotest:kotest-assertions-json:4.0.7")
     testImplementation("io.ktor:ktor-client-mock:$ktorVersion")
+    compileOnly("io.ktor:ktor-client-mock:$ktorVersion")
```
Comment (davidmotson) [inline common/build.gradle.kts:98]: nit: I would move this above the test dependencies, under implementation dependencies.

Comment (tanzimfh) [reply common/build.gradle.kts]: fixed, thanks!

File: .changes/generativeai/cellar-cobweb-bait-angle.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"type":"PATCH","changes":["Add test-only APIController constructor for Vertex AI unit tests"]}
```
Comment (rlazo) [inline .changes/generativeai/cellar-cobweb-bait-angle.json:1]: are there any changes to the `generativeai` sdk?

Comment (tanzimfh) [reply .changes/generativeai/cellar-cobweb-bait-angle.json:1]: nope, I just realized that `./change` adds a PATCH change when there are no changes at all. I'm deleting this file.

**Review by rlazo — APPROVED** [review]

**Pull request merged**